drivers/serial/uart_nrfx_uarte.c

```diff
@@ -909,8 +976,18 @@ static int uarte_nrfx_rx_enable(const struct device *dev, uint8_t *buf,
 		if (async_rx->flush_cnt) {
 			int cpy_len = MIN(len, async_rx->flush_cnt);
 
+			if (IS_ENABLED(UARTE_ANY_CACHE) &&
+			    (cfg->flags & UARTE_CFG_FLAG_CACHEABLE)) {
+				sys_cache_data_invd_range(cfg->rx_flush_buf, cpy_len);
```

Comment:
I guess this shouldn't be covered by `dmm`, as this is a case in which both `rx_flush_buf` and `buf` are located in cacheable RAM and in fact we are "emulating" DMA copying data from the former to the latter ?